conan/internal/conan_app.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, conan_api):
         """
         global_conf = conan_api.config.global_conf
         self.global_conf = global_conf
+        self.conan_api = conan_api
```

Comment:
Talked with @AbrilRBS about this. It looks better than changing more methods because we need to forward the hook manager to `run_validate_package_id`